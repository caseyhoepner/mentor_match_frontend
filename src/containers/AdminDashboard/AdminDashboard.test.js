import React from 'react';
import { shallow } from 'enzyme';
import { AdminDashboard } from './';
jest.mock('uuid', () => jest.fn(() => 1));

describe('AdminDashboard', () => {
  let wrapper;
  let mockMentors = [
    {
      name: 'Casey',
      preferences: {
        title: 'Awesome Coders'
      }
    },
    {
      name: 'Alex',
      preferences: {
        title: 'Other Coders'
      }
    }
  ]
  
  beforeEach(() => {
    wrapper = shallow(<AdminDashboard  mentors={mockMentors} />)
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});