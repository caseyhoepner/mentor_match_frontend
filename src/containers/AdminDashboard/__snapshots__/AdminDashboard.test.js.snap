// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminDashboard
    mentors={
      Array [
        Object {
          "name": "Casey",
          "preferences": Object {
            "title": "Awesome Coders",
          },
        },
        Object {
          "name": "Alex",
          "preferences": Object {
            "title": "Other Coders",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="ad-header"
        >
          <div
            className="ad-nav-btns"
          >
            <img
              alt="Turing Logo"
              className="ad-turing-logo"
              src="turing-logo.png"
            />
            <div
              className="ad-github-btn"
            >
              <p
                className="ad-gh-tagline"
              >
                Sign in with GitHub
              </p>
              <img
                alt="GitHub Logo"
                className="ad-github-logo"
                src={
                  Object {
                    "ReactComponent": [Function],
                    "__esModule": true,
                    "default": "github-logo.svg",
                  }
                }
              />
            </div>
          </div>
          <div
            className="ad-title"
          >
            <h1
              className="ad-title-text"
            >
              Admin Dashboard
            </h1>
            <div
              className="ad-line-break"
            />
            <p
              className="ad-tagline"
            >
              Filter mentors and students to find a match!
            </p>
          </div>
        </header>,
        <Connect(AdminMentorSearch) />,
        <div
          className="ad-mentor-headings"
        >
          <h1
            className="ad-mentor-heading"
          >
            Profile Picture
          </h1>
          <h1
            className="ad-mentor-heading"
          >
            Name
          </h1>
          <h1
            className="ad-mentor-heading"
          >
            Stack Preference
          </h1>
          <h1
            className="ad-mentor-heading"
          >
            Other Preferences
          </h1>
          <h1
            className="ad-mentor-heading"
          >
            View/Edit
          </h1>
        </div>,
        Array [
          <AdminMentorCard
            mentor={
              Object {
                "name": "Casey",
                "preferences": Object {
                  "title": "Awesome Coders",
                },
              }
            }
          />,
          <AdminMentorCard
            mentor={
              Object {
                "name": "Alex",
                "preferences": Object {
                  "title": "Other Coders",
                },
              }
            }
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ad-nav-btns"
            >
              <img
                alt="Turing Logo"
                className="ad-turing-logo"
                src="turing-logo.png"
              />
              <div
                className="ad-github-btn"
              >
                <p
                  className="ad-gh-tagline"
                >
                  Sign in with GitHub
                </p>
                <img
                  alt="GitHub Logo"
                  className="ad-github-logo"
                  src={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "github-logo.svg",
                    }
                  }
                />
              </div>
            </div>,
            <div
              className="ad-title"
            >
              <h1
                className="ad-title-text"
              >
                Admin Dashboard
              </h1>
              <div
                className="ad-line-break"
              />
              <p
                className="ad-tagline"
              >
                Filter mentors and students to find a match!
              </p>
            </div>,
          ],
          "className": "ad-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Turing Logo"
                  className="ad-turing-logo"
                  src="turing-logo.png"
                />,
                <div
                  className="ad-github-btn"
                >
                  <p
                    className="ad-gh-tagline"
                  >
                    Sign in with GitHub
                  </p>
                  <img
                    alt="GitHub Logo"
                    className="ad-github-logo"
                    src={
                      Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "github-logo.svg",
                      }
                    }
                  />
                </div>,
              ],
              "className": "ad-nav-btns",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Turing Logo",
                  "className": "ad-turing-logo",
                  "src": "turing-logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="ad-gh-tagline"
                    >
                      Sign in with GitHub
                    </p>,
                    <img
                      alt="GitHub Logo"
                      className="ad-github-logo"
                      src={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "github-logo.svg",
                        }
                      }
                    />,
                  ],
                  "className": "ad-github-btn",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sign in with GitHub",
                      "className": "ad-gh-tagline",
                    },
                    "ref": null,
                    "rendered": "Sign in with GitHub",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "GitHub Logo",
                      "className": "ad-github-logo",
                      "src": Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "github-logo.svg",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="ad-title-text"
                >
                  Admin Dashboard
                </h1>,
                <div
                  className="ad-line-break"
                />,
                <p
                  className="ad-tagline"
                >
                  Filter mentors and students to find a match!
                </p>,
              ],
              "className": "ad-title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Admin Dashboard",
                  "className": "ad-title-text",
                },
                "ref": null,
                "rendered": "Admin Dashboard",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ad-line-break",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Filter mentors and students to find a match!",
                  "className": "ad-tagline",
                },
                "ref": null,
                "rendered": "Filter mentors and students to find a match!",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="ad-mentor-heading"
            >
              Profile Picture
            </h1>,
            <h1
              className="ad-mentor-heading"
            >
              Name
            </h1>,
            <h1
              className="ad-mentor-heading"
            >
              Stack Preference
            </h1>,
            <h1
              className="ad-mentor-heading"
            >
              Other Preferences
            </h1>,
            <h1
              className="ad-mentor-heading"
            >
              View/Edit
            </h1>,
          ],
          "className": "ad-mentor-headings",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Profile Picture",
              "className": "ad-mentor-heading",
            },
            "ref": null,
            "rendered": "Profile Picture",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name",
              "className": "ad-mentor-heading",
            },
            "ref": null,
            "rendered": "Name",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Stack Preference",
              "className": "ad-mentor-heading",
            },
            "ref": null,
            "rendered": "Stack Preference",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Other Preferences",
              "className": "ad-mentor-heading",
            },
            "ref": null,
            "rendered": "Other Preferences",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "View/Edit",
              "className": "ad-mentor-heading",
            },
            "ref": null,
            "rendered": "View/Edit",
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "mentor": Object {
            "name": "Casey",
            "preferences": Object {
              "title": "Awesome Coders",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "mentor": Object {
            "name": "Alex",
            "preferences": Object {
              "title": "Other Coders",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="ad-header"
          >
            <div
              className="ad-nav-btns"
            >
              <img
                alt="Turing Logo"
                className="ad-turing-logo"
                src="turing-logo.png"
              />
              <div
                className="ad-github-btn"
              >
                <p
                  className="ad-gh-tagline"
                >
                  Sign in with GitHub
                </p>
                <img
                  alt="GitHub Logo"
                  className="ad-github-logo"
                  src={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "github-logo.svg",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="ad-title"
            >
              <h1
                className="ad-title-text"
              >
                Admin Dashboard
              </h1>
              <div
                className="ad-line-break"
              />
              <p
                className="ad-tagline"
              >
                Filter mentors and students to find a match!
              </p>
            </div>
          </header>,
          <Connect(AdminMentorSearch) />,
          <div
            className="ad-mentor-headings"
          >
            <h1
              className="ad-mentor-heading"
            >
              Profile Picture
            </h1>
            <h1
              className="ad-mentor-heading"
            >
              Name
            </h1>
            <h1
              className="ad-mentor-heading"
            >
              Stack Preference
            </h1>
            <h1
              className="ad-mentor-heading"
            >
              Other Preferences
            </h1>
            <h1
              className="ad-mentor-heading"
            >
              View/Edit
            </h1>
          </div>,
          Array [
            <AdminMentorCard
              mentor={
                Object {
                  "name": "Casey",
                  "preferences": Object {
                    "title": "Awesome Coders",
                  },
                }
              }
            />,
            <AdminMentorCard
              mentor={
                Object {
                  "name": "Alex",
                  "preferences": Object {
                    "title": "Other Coders",
                  },
                }
              }
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ad-nav-btns"
              >
                <img
                  alt="Turing Logo"
                  className="ad-turing-logo"
                  src="turing-logo.png"
                />
                <div
                  className="ad-github-btn"
                >
                  <p
                    className="ad-gh-tagline"
                  >
                    Sign in with GitHub
                  </p>
                  <img
                    alt="GitHub Logo"
                    className="ad-github-logo"
                    src={
                      Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "github-logo.svg",
                      }
                    }
                  />
                </div>
              </div>,
              <div
                className="ad-title"
              >
                <h1
                  className="ad-title-text"
                >
                  Admin Dashboard
                </h1>
                <div
                  className="ad-line-break"
                />
                <p
                  className="ad-tagline"
                >
                  Filter mentors and students to find a match!
                </p>
              </div>,
            ],
            "className": "ad-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Turing Logo"
                    className="ad-turing-logo"
                    src="turing-logo.png"
                  />,
                  <div
                    className="ad-github-btn"
                  >
                    <p
                      className="ad-gh-tagline"
                    >
                      Sign in with GitHub
                    </p>
                    <img
                      alt="GitHub Logo"
                      className="ad-github-logo"
                      src={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "github-logo.svg",
                        }
                      }
                    />
                  </div>,
                ],
                "className": "ad-nav-btns",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Turing Logo",
                    "className": "ad-turing-logo",
                    "src": "turing-logo.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="ad-gh-tagline"
                      >
                        Sign in with GitHub
                      </p>,
                      <img
                        alt="GitHub Logo"
                        className="ad-github-logo"
                        src={
                          Object {
                            "ReactComponent": [Function],
                            "__esModule": true,
                            "default": "github-logo.svg",
                          }
                        }
                      />,
                    ],
                    "className": "ad-github-btn",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sign in with GitHub",
                        "className": "ad-gh-tagline",
                      },
                      "ref": null,
                      "rendered": "Sign in with GitHub",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "GitHub Logo",
                        "className": "ad-github-logo",
                        "src": Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "github-logo.svg",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="ad-title-text"
                  >
                    Admin Dashboard
                  </h1>,
                  <div
                    className="ad-line-break"
                  />,
                  <p
                    className="ad-tagline"
                  >
                    Filter mentors and students to find a match!
                  </p>,
                ],
                "className": "ad-title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Admin Dashboard",
                    "className": "ad-title-text",
                  },
                  "ref": null,
                  "rendered": "Admin Dashboard",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "ad-line-break",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Filter mentors and students to find a match!",
                    "className": "ad-tagline",
                  },
                  "ref": null,
                  "rendered": "Filter mentors and students to find a match!",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="ad-mentor-heading"
              >
                Profile Picture
              </h1>,
              <h1
                className="ad-mentor-heading"
              >
                Name
              </h1>,
              <h1
                className="ad-mentor-heading"
              >
                Stack Preference
              </h1>,
              <h1
                className="ad-mentor-heading"
              >
                Other Preferences
              </h1>,
              <h1
                className="ad-mentor-heading"
              >
                View/Edit
              </h1>,
            ],
            "className": "ad-mentor-headings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Profile Picture",
                "className": "ad-mentor-heading",
              },
              "ref": null,
              "rendered": "Profile Picture",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
                "className": "ad-mentor-heading",
              },
              "ref": null,
              "rendered": "Name",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Stack Preference",
                "className": "ad-mentor-heading",
              },
              "ref": null,
              "rendered": "Stack Preference",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Other Preferences",
                "className": "ad-mentor-heading",
              },
              "ref": null,
              "rendered": "Other Preferences",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "View/Edit",
                "className": "ad-mentor-heading",
              },
              "ref": null,
              "rendered": "View/Edit",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "mentor": Object {
              "name": "Casey",
              "preferences": Object {
                "title": "Awesome Coders",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "mentor": Object {
              "name": "Alex",
              "preferences": Object {
                "title": "Other Coders",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
